sdk/python/kfp/dsl/_container_op.py

```diff
@@ -1371,6 +1383,64 @@ def add_node_selector_constraint(self, label_name: str,
     super(ContainerOp, self).add_node_selector_constraint(label_name, value)
     return self
 
+  def add_cpu_request(self, cpu: Union[str,  _pipeline_param.PipelineParam]) -> 'ContainerOp':
+    """Adds a cpu request at runtime.
+
+    Args:
+        cpu (Union[str, PipelineParam]): kubernetes cpu request,
+        https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource/#specify-a-cpu-request-and-a-cpu-limit
+    """
+
+    self._validate_cpu_string(cpu)
+    self._cpu_request = cpu
+    return self
+
+  def add_memory_request(self, memory: Union[str,  _pipeline_param.PipelineParam]) -> 'ContainerOp':
+    """Adds a memory request at runtime.
+
+    Args:
+        memory (Union[str, PipelineParam]): kubernetes memory request, 
+        https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-memory
+    """
+
+    self._validate_size_string(memory)
+    self._memory_request = memory
+    return self
+
+  def _validate_size_string(self, size_string):
```

Comment:
Looks like this is from the existing code, can we reuse the existing one? (may need some refactor?)